### PR TITLE
Multiple source directories code review

### PR DESCRIPTION
**Pull request by sentience**
Code quality improvements following post-merge comments on NoRedInk/find-elm-dependencies#1 (thanks again, @eeue56!).

### CODE REVIEW

**Comment by sentience** [comment]
@eeue56 Just making sure this is on your radar.

File: index.js [hunk 1]
```diff
@@ -98,7 +99,13 @@ function isRoot(dir) {
 }
 
 function getSourceDirectories(elmPackagePath) {
-  var elmPackage = JSON.parse(fs.readFileSync(elmPackagePath, 'utf8'));
+  try {
+    var elmPackage = JSON.parse(fs.readFileSync(elmPackagePath, 'utf8'));
+  } catch (e) {
+    // invalid JSON
+    return [];
```
Comment (eeue56) [inline index.js]: probably worth logging an error message as well as this

Comment (sentience) [reply index.js]: @eeue56 Done!

**Review by eeue56 — APPROVED** [review]

**Comment by sentience** [comment]
@stoeffel Are you happy to publish this?

File: index.js [hunk 2]
```diff
@@ -98,7 +99,13 @@ function isRoot(dir) {
 }
 
 function getSourceDirectories(elmPackagePath) {
-  var elmPackage = JSON.parse(fs.readFileSync(elmPackagePath, 'utf8'));
+  try {
+    var elmPackage = JSON.parse(fs.readFileSync(elmPackagePath, 'utf8'));
```
Comment (stoeffel) [inline index.js]: maybe we should also check if `elmPackage` has a `source-directories` field, otherwise it will crash on line 109

**Review by stoeffel — CHANGES_REQUESTED** [review]
@sentience LGTM, but added one last comment.

File: index.js (hunk 2 above)
Comment (sentience) [reply index.js]: Agreed. All done!

**Review by stoeffel — APPROVED** [review]

File: index.js [hunk 3]
```diff
@@ -106,7 +106,13 @@ function getSourceDirectories(elmPackagePath) {
     return [];
   }
 
-  return elmPackage['source-directories'].map(function(sourceDir) {
+  var sourceDirectories = elmPackage['source-directories'];
+  if (!Array.isArray(sourceDirectories)) {
+    console.warn('Ignored package metadata JSON file with missing/invalid source-directories: ' + elmPackagePath);
```
Comment (stoeffel) [inline index.js]: 😻

**Comment by stoeffel** [comment]
published with v1.0.1

**Pull request merged**